core/trace/dt_span_serializer.go

```diff
@@ -136,6 +138,8 @@ func createProtoSpan(dtSpan *dtSpan, incomingCustomTag *protoTrace.CustomTag) (*
 			spanMsg.CustomTag = incomingCustomTag
 		}
 
+		encodedLinkId := spanMetadata.fw4Tag.EncodedLinkID()
+		spanMsg.ParentFwtagEncodedLinkId = &encodedLinkId
```

Comment:
I don't see any IsRemote check for the link ID. Relatedly, is fw4Tag here always non-nil?

Reply:
The fw4tag will always be non-nil, each `dtSpanMetadata` must have an fw4tag. 

I've moved these two lines inside a `parentSpanCtx.IsRemote()` check